src/tests/itests/xi_itests.c

```diff
@@ -12,20 +12,21 @@
 #include "xi_itest_tls_layer.h"
 #endif
 #include "xi_itest_mqttlogic_layer.h"
+#include "xi_itest_update.h"
 #undef XI_MOCK_TEST_PREPROCESSOR_RUN
 
 #include "xi_test_utils.h"
 #include "xi_lamp_communication.h"
 
-struct CMGroupTest groups[] = {
-    cmocka_test_group( xi_itests_clean_session ), cmocka_test_group( xi_itests_tls_error )
+struct CMGroupTest groups[] = {cmocka_test_group( xi_itests_clean_session ),
+                               cmocka_test_group( xi_itests_tls_error ),
 #ifndef XI_NO_TLS_LAYER
-                                                      ,
-    cmocka_test_group( xi_itests_tls_layer )
+                               cmocka_test_group( xi_itests_tls_layer ),
 #endif
-        ,
-    cmocka_test_group( xi_itests_mqttlogic_layer ),
-    cmocka_test_group( xi_itests_connect_error ), cmocka_test_group_end};
+                               cmocka_test_group( xi_itests_mqttlogic_layer ),
+                               cmocka_test_group( xi_itests_connect_error ),
+                               cmocka_test_group( xi_itests_update ),
```

Comment:
Agree.
- device_update?
- file_update?
- firmware_update isn't good since it is not "just" firmware update.
- sft isn't enough either, since it is not only about secure file transfer, it is about maintaining the files and even applying an action on the file / device.